main.c

```diff
@@ -200,6 +200,16 @@ MPZ_from_str(PyObject *s, int base)
     if (base > 36) {
         digit_value += 208;
     }
+    if (base == 10) {
```

Comment:
There is nothing special in this base.

Try to understand how the function works.  When you do this, fix will be obvious.  It's really just one-line check.

Reply:
I will read it again. It's really hard to understand, I will try my best.